From cc15596b31dbb74b6dca86f28a01a41c2cab51df Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Thu, 26 Oct 2017 17:19:15 +0300
Subject: [PATCH 0577/1051] irqchip/gicv3-its: mvebu: update cache &
 sharability for ITS tables

Update cache and sharability for ITS tables when MVEBU
workaroud is enabled and set them as non cachable & non sharable,
because of bug in fabric connection.

Change-Id: I7b89a7d0b612709bfd81119f7851f3ce9fb6d324
Signed-off-by: Hanna Hawa <hannah@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06.
Just some minor context mods in order to port to wrlinux]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/irqchip/irq-gic-v3-its.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/irqchip/irq-gic-v3-its.c b/drivers/irqchip/irq-gic-v3-its.c
index e86ab30f6137..9d928a6ed714 100644
--- a/drivers/irqchip/irq-gic-v3-its.c
+++ b/drivers/irqchip/irq-gic-v3-its.c
@@ -1808,6 +1808,11 @@ static int its_alloc_tables(struct its_node *its)
 		/* erratum 24313: ignore memory access type */
 		cache = GITS_BASER_nCnB;
 
+	if (its->flags & ITS_FLAGS_WORKAROUND_MVEBU_ERRATA) {
+		shr = GITS_BASER_NonShareable;
+		cache = GITS_BASER_nCnB;
+	}
+
 	for (i = 0; i < GITS_BASER_NR_REGS; i++) {
 		struct its_baser *baser = its->tables + i;
 		u64 val = its_read_baser(its, baser);
-- 
2.17.1

